actions/pull-request/download-artifact/entrypoint/main_test.go

```diff
@@ -0,0 +1,254 @@
+					_, _ = w.Write(buf.Bytes())
+
+				case "/repos/some-owner/nonexistent-repo/actions/runs/45678/artifacts":
+					w.WriteHeader(http.StatusNotFound)
+
+				case "/repos/some-owner/some-repo/actions/artifacts/55555/zip":
+					filename := "another-payload"
+					w.Header().Set("Content-Type", "application/json")
+					w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", filename))
+					w.WriteHeader(http.StatusOK)
+
+				default:
+					t.Fatal(fmt.Sprintf("unknown path: %s", req.URL.Path))
+				}
+			}))
+
+			tempDir, err = os.MkdirTemp("", "output")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(os.Setenv("GITHUB_WORKSPACE", tempDir)).To(Succeed())
+		})
+
+		it.After(func() {
+			mockServer.Close()
+			Expect(os.Unsetenv("GITHUB_WORKSPACE")).To(Succeed())
+			Expect(os.RemoveAll(tempDir)).To(Succeed())
+		})
+
+		context("given arguments that point to a repo with an artifact", func() {
+			it("gets the artifact zip and unzips it onto the file system", func() {
+				command := exec.Command(
+					entrypoint,
+					"--name", "payload",
+					"--repo", "some-owner/some-repo",
+					"--runID", "12345",
+					"--githubAPI", mockServer.URL,
+				)
+
+				buffer := gbytes.NewBuffer()
+				session, err := gexec.Start(command, buffer, buffer)
+				Expect(err).NotTo(HaveOccurred())
+
+				Eventually(session).Should(gexec.Exit(0), func() string { return string(buffer.Contents()) })
+
+				out := string(buffer.Contents())
+				Expect(out).To(ContainLines(
+					"Getting workflow artifacts",
+					"Getting workflow artifact zip file",
+					"Reading file: payload",
+				))
+
+				contents, _ := os.ReadFile(filepath.Join(tempDir, "event.json"))
+				Expect(string(contents)).To(MatchJSON(`{
+				"action": "synchronize",
+				"pull_request": {
+					"_links": {
+						"comments": {
+							"href": "https://api.github.com/repos/some-org/some-repo/issues/1/comments"
+						},
+						"commits": {
+							"href": "https://api.github.com/repos/some-org/some-repo/pulls/1/commits"
+						}
+					},
+					"body": "Body of PR",
+					"changed_files": 2,
+					"closed_at": null,
+					"comments": 0,
+					"commits": 5,
+					"deletions": 52,
+					"labels": [],
+					"number": 1,
+					"state": "open",
+					"title": "Title",
+					"user": {
+						"id": 98765,
+						"login": "paketo-bot"
+					}
+				}
+				}`))
+			})
+		})
+
+		context("failure cases", func() {
+			context("no workflow artifact exists", func() {
+				it("returns an error and exits non-zero", func() {
+					command := exec.Command(
+						entrypoint,
+						"--name", "payload",
+						"--repo", "some-owner/nonexistent-repo",
+						"--runID", "45678",
+						"--githubAPI", mockServer.URL,
+					)
+
+					buffer := gbytes.NewBuffer()
+					session, err := gexec.Start(command, buffer, buffer)
+					Expect(err).NotTo(HaveOccurred())
+
+					Eventually(session).Should(gexec.Exit(1), func() string { return string(buffer.Contents()) })
```

Comment:
I thought that `Eventually(session).Should(gexec.Exit(1))` would give the session output if the assertion failed. Did you find that this was not happening?

Reply:
@ryanmoran I'm not sure what you mean by this. I modeled the tests off of the jam update-builder tests which don't do anything regarding the session output, so I hadn't really looked into it.  Do you mean looking at `session.Out` bytes for the error message, rather than looking at the `buffer.Contents`?

Reply:
Don't worry about it. It doesn't work the way I expected. This code looks fine.